### DISCUSSION TITLE
Can python-oracledb be used instead of cx_Oracle?

### DISCUSSION THREAD

**Discussion by GreenReaper** [pretalx/pretalx#1411] [Self-hosting] [answered]
The [pretalx configuration documentation](https://docs.pretalx.org/administrator/configure.html#the-database-section) specifies [cx_Oracle](https://oracle.github.io/python-cx_Oracle/) for use to interface with Oracle databases. Last year Oracle released a slimmed-down version [python-oracledb](https://github.com/oracle/python-oracledb) that connects directly without the client libraries (https://github.com/oracle/python-cx_Oracle/issues/628).

Is python-oracledb sufficient? Has anyone tried this? I have an Oracle Cloud account and think Pretalx might actually be a good use for it, since it comes with two 20GB DB instances for free, but I don't know if any of the [thick-mode functionality](https://python-oracledb.readthedocs.io/en/latest/user_guide/appendix_a.html#featuresummary) is required by pretalx.

**Comment by rixx** [comment, marked as answer]
That's a good question – honestly, we don’t test on Oracle at all, and I have never deployed pretalx on it. I’ll remove that section of the docs. Pretending we support a database we don’t even test strikes me as pretty bad practice. (Sorry, I imagine that this is not the response you wanted to hear!)

**Reply by GreenReaper** [reply]
Honestly, that's a fine response. I don't know how much of an abstraction Python's data layer is; Oracle is somewhat like PostgreSQL if not entirely, so at least there's that - many projects are MySQL/MariaDB first and I imagine they might be even less compatible. If I ever try it I'll report back, since as noted above it might be a good use of the Free Tier.

**Reply by rixx** [reply]
Chances are good that things are going to work! Django should provide a working abstraction for all the things we need, as we continue to support SQLite + MariaDB + Postgres, with no particular preference, which Django also smoothes over. But I honestly can't be bothered to set up tests for a database that has not been used so far (I'd probably merge a PR adding it to the test matrix? Not sure …).
But yeah, please do report back if you end up attempting it – I probably wouldn't advertise it either way, but it would be good to know!
